File: InsertionSort/InsertionSort3.c
#include<stdio.h>

void insertionSort(int *data,int size)
{
int e,f,num,m;
m=size-1;
for(e=1;e<=m;e++)
{
for(num=data[e],f=e-1;f>=0 && data[f]>num;f--) data[f+1]=data[f];
data[f+1]=num;
}
}

int main()
{
int x[5];
int y,m;
m=5-1;
for(y=0;y<=m;y++)
{
printf("Enter a number : ");
scanf("%d",&x[y]);
fflush(stdin);
}
insertionSort(x,5);
for(y=0;y<=m;y++) printf("%d\n",x[y]);
return 0;
}

File: InsertionSort/InsertionSort3.java
import java.io.*;
class InsertionSort3
{
public static void insertionSort(int x[])
{
int e,f,m,num;
m=x.length-1;
e=1;
while(e<=m)
{
num=x[e];
f=e-1;
while(f>=0 && x[f]>num)
{
x[f+1]=x[f];
f--;
}
x[f+1]=num;
e++;
}
}
public static void main(String args[])
{
BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
int x[],y,e,f,m,req,num;
req=0;
try
{
System.out.print("Enter your requirement : ");
req=Integer.parseInt(br.readLine());
}catch(IOException ioException)
{
System.out.println(ioException);
}catch(NumberFormatException nfe)
{
System.out.println(nfe);
}
m=req-1;
x=new int[req];
for(y=0;y<=m;y++)
{
System.out.print("Enter a number : ");
try
{
x[y]=Integer.parseInt(br.readLine());
}catch(IOException ioException)
{
System.out.println(ioException);
}catch(NumberFormatException nfe)
{
System.out.println(nfe);
}
}
insertionSort(x);
for(y=0;y<=m;y++) System.out.println(x[y]);
}
}
